backend/scripts/calculateAirdrop/getFinalBalances.ts

```diff
@@ -0,0 +1,73 @@
+import { task } from "hardhat/config";
+import fs from "fs";
+import path from "path";
+import { getLiqSnapshotsForPair } from "../../subgraph/fetchers";
+import { BigNumber } from "ethers";
+import { AirdropAmounts, LiqSnapshot, UserData } from ".";
+
+task(
+  "get-final-balances",
+  "Final balance at the end, taking into account inward and outward transfers after the snapshot period"
+)
+  .addParam("blocknumber", "Block Number")
+  .addParam("endblocknumber", "End Block Number")
+  .addVariadicPositionalParam("addressesArray")
+  .setAction(
+    async (
+      { blocknumber, addressesArray, endblocknumber },
+      { network, ethers }
+    ) => {
+      const filePath = path.resolve(
+        __dirname,
+        "output/userBalancesSnapshot.json"
+      );
+      const userData = JSON.parse(fs.readFileSync(filePath).toString());
+
+      let allBalances: AirdropAmounts = {};
+
+      for (var index = 0; index < addressesArray.length; index++) {
+        const { users, balances }: UserData = userData[addressesArray[index]];
+        const allLiqSnapshots = await getLiqSnapshotsForPair(
+          network.name,
+          blocknumber,
+          endblocknumber,
+          addressesArray[index]
+        );
+        allBalances[addressesArray[index]] = {};
+        users.forEach((user, jindex) => {
+          if (user !== "0x0000000000000000000000000000000000000000") {
+            let userFinalBalance = BigNumber.from(balances[jindex]);
+            let overflow = BigNumber.from(0);
+            allLiqSnapshots.forEach((snapshot: LiqSnapshot) => {
+              if (snapshot.user.id === user) {
+                const liqBalanceBN = ethers.utils.parseEther(
+                  snapshot.liquidityTokenBalance
+                );
+                const gaugeBalanceBN = ethers.utils.parseEther(
+                  snapshot.gaugeBalance
+                );
+                let delta = userFinalBalance
+                  .mul(-1)
+                  .add(liqBalanceBN)
+                  .add(gaugeBalanceBN);
+                overflow = overflow.add(delta);
+                if (overflow.lt(0)) {
+                  userFinalBalance = userFinalBalance.add(overflow);
+                  overflow = BigNumber.from("0");
+                }
+              }
```

Comment:
I understood the logic, but I am not seeing it being applied here.
Here is a scenario:
```
user balance = 1000
1 - the balance snapshot is 700
  delta = -1000 + 700 = -300
  overflow = 0 - 300 = -300
  userfinalbalance = 1000 - 300 = 700
  overflow = 0
2 - the balance snapshot is 1200
  delta = -700 + 1200 = 500
  overflow = 0 + 500 = 500
  overflow being >= 0 then we stop here
3 - no more snapshot so we leave the loop
4 - balance will be 700 whereas the overflow is positive and has never been counted in the user balance
```

Is this case valid or should we always count a positive overflow?